internal/grpc/services/sharesstorageprovider/sharesstorageprovider.go

```diff
@@ -585,24 +575,23 @@ func (s *service) Move(ctx context.Context, req *provider.MoveRequest) (*provide
 	//      - the registry needs to invalidate the alias
 	//      - the rhe share manager needs to change the name
 	//      ... but which storageprovider will receive the move request???
-	srcReceivedShare, rpcStatus, err := s.resolveReference(ctx, req.Source)
+	srcReceivedShare, rpcStatus, err := s.resolveAcceptedShare(ctx, req.Source)
 	if err != nil {
 		return nil, err
 	}
-	if rpcStatus != nil {
+	if rpcStatus.Code != rpc.Code_CODE_OK {
 		return &provider.MoveResponse{
 			Status: rpcStatus,
 		}, nil
 	}
 
-	// can we do a rename
-	if utils.ResourceIDEqual(req.Source.ResourceId, req.Destination.ResourceId) &&
-		// only if we are responsible for the space
-		req.Source.ResourceId.StorageId == utils.ShareStorageProviderID &&
-		// only if the source path has no path segment
-		req.Source.Path == "." &&
-		// only if the destination is a dot followed by a single path segment, e.g. './new'
-		len(strings.SplitN(req.Destination.Path, "/", 3)) == 2 {
+	// we can do a rename
+	// if the source is a share jail child where the path is .
+	if ((isShareJailChild(req.Source.ResourceId) && req.Source.Path == ".") ||
+		// or if the source is the share jail with a single path segment, e.g. './old'
+		(isShareJailRoot(req.Source.ResourceId) && len(strings.SplitN(req.Destination.Path, "/", 3)) == 2)) &&
+		// and if the destination is a dot followed by a single path segment, e.g. './new'
+		isShareJailRoot(req.Destination.ResourceId) && len(strings.SplitN(req.Destination.Path, "/", 3)) == 2 {
```

Comment:
It's pretty hard to wrap my head around this, but it looks to me as if there is an error. Line 592 should check for `req.Source.Path`.

Any chance we can get this a little bit clearer? Maybe using a dedicated func?

Reply:
just a little bit